ui/lib/kv/addon/components/page/secret/details.js

```diff
@@ -14,19 +14,50 @@ import { tracked } from '@glimmer/tracking';
  *  @secretPath={{this.model.path}}
  *  @secret={{this.model.secret}}
  *  @metadata={{this.model.metadata}}
+ *  @noMetadataPermission={{eq this.model.metadata 403}}
  *  @breadcrumbs={{this.breadcrumbs}}
   /> 
  *
  * @param {string} secretPath - path of kv secret 'my/secret' used as the title for the KV page header 
  * @param {model} secret - Ember data model: 'kv/data'  
  * @param {model} metadata - Ember data model: 'kv/metadata'
+ * @param {boolean} noMetadataPermission - True if we received a 403 from the kv/metadata network request
  * @param {array} breadcrumbs - Array to generate breadcrumbs, passed to the page header component
  */
 
 export default class KvSecretDetails extends Component {
   @tracked showJsonView = false;
+
   @action
   toggleJsonView() {
     this.showJsonView = !this.showJsonView;
   }
+
+  get emptyState() {
+    const { version, canReadData, destroyed, deletionTime } = this.args.secret;
+    if (!canReadData) {
+      return {
+        title: 'You do not have permission to read this secret',
+        message:
+          'Your policies may permit you to write a new version of this secret, but do not allow you to read its current contents.',
+      };
+    }
+    if (destroyed) {
+      return {
+        title: `Version ${version} of this secret has been permanently destroyed`,
+        message:
+          'A version that has been permanently deleted cannot be restored. You can view other versions of this secret in the Version History tab above.',
+        link: '/vault/docs/secrets/kv/kv-v2',
+      };
+    }
+    if (deletionTime) {
+      return {
+        title: `Version ${version} of this secret has been deleted`,
+        message:
+          'This version has been deleted but can be undeleted. View other versions of this secret by clicking the Version History tab above.',
```

Comment:
So the undeleted option isn't for everyone. It comes down to permissions. Maybe amend to clarify that they'll need a policy that allows them to [undelete](https://developer.hashicorp.com/vault/api-docs/secret/kv/kv-v2#undelete-secret-versions)?

Reply:
Also, sorry another permissions thing. A lot of folks that can view a secret and even undelete a secret wont' see the version history option. They won't have access to metadata. So maybe leave that sentence out all together?

Reply:
I was just copying this text from the old code - but changing the copy is part of this epic?

Reply:
Oish. dang. I can't believe that's old copy (well, okay I can). No nothing new is part of this. It's you're call. If you would rather not do any copy change, I understand, but if you think it's minor and would clarify then go for it.

Reply:
I think it might be worth doing a copy audit at the end once we've gotten all the states cleared up?